pandas/tests/frame/test_query_eval.py

```diff
@@ -1176,3 +1176,17 @@ def test_failing_character_outside_range(self, df):
     def test_failing_hashtag(self, df):
         with pytest.raises(SyntaxError):
             df.query("`foo#bar` > 4")
+
+    def test_call_non_named_expression(self, df):
+        def func(*_):
+            return 1
+
+        funcs = [func]  # noqa
+
+        df.eval("@func()")
+
+        with pytest.raises(TypeError):
```

Comment:
can you test the exception message as well, it is not clear what the user is going to see here